src/utils/openCloseComponent.ts

```diff
@@ -66,8 +71,9 @@ function transitionEnd(event: TransitionEvent): void {
  * Used to emit (before)open/close events both for when the opacity transition is present and when there is none (transition-duration is set to 0).
  *
  * @param component
+ * @param nonOpenCloseComponent
  */
-export function onToggleOpenCloseComponent(component: OpenCloseComponent): void {
+export function onToggleOpenCloseComponent(component: OpenCloseComponent, nonOpenCloseComponent = false): void {
   readTask((): void => {
```

Comment:
Is there any other `transitionProp` we expect to handle besides `open` and `expanded`? If not, maybe we can name it `expandedCollapsedComponent` instead of `nonOpenCloseComponent` in here? And check for `expanded` in the ternary below... Or we could at some point refactor/rename the util and have one `transitionProp` which we set from the getgo.

Reply:
i don't have a strict preference here. the reason behind is to convey the idea of deviating from the original naming pattern `openClose` . This might help if we have another set of `transitionProp` like expanded in some other component or we can refactor the interface to something like `transitionComponent`.